Post sobre disfunção executiva (#22)
danmrc/azul: content/post/26-09-2022-o-cerebro-os-millenials-e-a-homeostase/post.Rmd

# Níveis de Marr

Note que estamos rapidamente transitando entre escalas diferente, quase que por obrigação. Em um momento falamos de algo visível a olho nu, depois em células únicas, por vezes de sistemas inteiros, por vezes de propriedades matemáticas de certos algoritmos. Essa é uma fonte de problemas, é muito fácil começar uma frase em um contexto e terminar em outro, se confundindo no caminho. Uma Teoria de Consciência precisa, simultaneamente, explicar fenômenos em escalas espaciais, temporais e computacionais totalmente distintas e precisamos de algum _framework_ para evitar confusões básicas. @marr1979computational oferece uma maneira de quebrar nossas perguntas (e respostas) em escopos distintos que conversam naturalmente com nossos métodos. Nesse framework, qualquer sistema com aparente processamento de informação pode ser entendido em 3 níveis:

* Funcional/Computacional: Quais são as tarefas que esse sistema realiza? Que tipo de comportamento é implementado, quais são seus modos de falha, quais são as origens evolutivas?

> * Algoritmico/Representacional: De que maneira essas funções são representadas e implementadas, quais são os algoritmos usamos para que elas sejam computadas?

Author: ```suggestion
* Algoritmico/Representacional: De que maneira essas funções são representadas e implementadas, quais são os algoritmos usados na sua computação?
```

* Implementação: Quais mecanismos físicos são recrutados para persistir a representação e executar os algoritmos em questão?

Para tangibilizar isso um pouco, vamos imaginar um Large Language Model (LLM) treinado em um grande corpus de texto. Ela tem habilidades verbais de um jovem razoavelmente lido, um humor seco, referências literárias e passa em alguma versão fraca do teste de Turing (não que isso seja grandes coisas). Em um nível _funcional_, não existe uma maneira simples de distinguir o que essa rede neural faz do que um humano faz. Descendo um andar, aparecem as distinções. 

> As representações e algoritmos de uma rede neural são radicalmenete diferentes das representações e algoritmos que sistemas nervosos usam para funcionar. Isso não demanda mais explicações a menos é claro que você, leitor, tenha uma certa fé na analogia Cérebro - Rede Neural.

Author: ```suggestion
As representações e algoritmos de uma rede neural são radicalmenete diferentes das representações e algoritmos que sistemas nervosos usam para funcionar. Isso não demanda mais explicações a menos é claro que você, leitor, tenha uma certa fé na analogia Cérebro - Rede Neural. As distinções de implementação mais ainda, rodamos redes neurais em circuitos digitais, nossos cérebros se aproveitam de várias propriedades do campo eletromagnético par encodar informação na topologia de rede do sistema nervoso.
```

> # Anilhamento, Entropia, Criticalidade, Organização

Author: ```suggestion
# Anilhamento, Entropia, Criticalidade, Organização

*Pausa dramática*. Eu sou um músico amador. _Como músico amador_, é difícil articular que diferença faz aprender a tocar "piano instrumental". E por isso quero dizer o tipo de coisa que cobrariam em vestibular de física na UERJ, se estudar música fosse obrigatório no ensino médio. O mínimo que um físico precisa saber de música. Algo análogo a teoria de conjuntos, geometria euclidiana e probabilidade de ensino médio. A escala maior e os arpeggios dos acordes diatônicos da escala maior. Nada demais, tenho certeza que os músicos lendo isso concordam.

E é inacreditável o quanto dá para "enganar" apenas sabendo a escala maior em alguns tons diferentes e os arpeggios dos acordes diatônicos. Um mundo de possibilidades e sons se abre, o inefável da música se torne legível. Acho que isso acontece porque (i) são dados nomes à fenômenos complicados e (ii) o piano é o resultado de quase um milênio de P&D em instrumentos musicais para o *estudo* da música. Quem toca guitarra, baixo, saxofone, bateria, etc, faz porque ama o timbre. A maioria das pessoas que tocam piano fazem não pelo piano em si, mas pelo seu design. Apenas baseado na interface do piano, você deve achar que Dó Bemol e Si são a mesma nota. Essa é uma simplificação útil. 

Bem, até agora eu me ative a o que julgo serem _fatos incontroversos_ da ciência aplicada, e um pouco de bom gosto (que se aplicado com rigor o suficiente dá para chamar de filosofia). Apenas fiz uma colagem deles. Agora vou dar um passo além e argumentar algo mais forte. Me chame de louco, mas a Evolução não teria chegado em desenhos não-triviais e interessantes se ela precisa, em paralelo, resolver problemas em escalas diferentes. Me parece que a evolução, muito antes de Turing, descobriu que com algum truques matemáticos como descida de gradiente, tabelas-verdade, branching condicional, era possível representar o processo de _navegar espaços de problemas_. Uma previsão que vem daqui então é que: cérebros e sistemas nervosos provavelmente dependem de alguns poucos princípios de auto-organização, que são aplicados universalmente em problemas diferentes.

```

Nossa interface com a realidade, o que ouvimos, o que vemos, os sabores que sentimos, as texturas que tocamos, está sujeita às mesmas limitações da visão. Captamos estímulos com variados sensores, que os traduzem em sinais elétricos, que são usados por sistemas especialistas no cérebro para compor uma "renderização", feita para ser utilizada por outros sistemas especialistas no cérebro. Vivemos em uma alucinação controlada - isso é resumido no experimento mental do [Cérebro numa Cuba](https://plato.stanford.edu/Archives/spr2009/entries/brain-vat/).

Lembra do Fisicalismo e do Qualia-Formalismo lá de cima? Adiciono uma terceira perna:

* Realismo Indireto: Existe uma Realidade comum e nossa percepção dela é mediada por processos computacionais que traduzem estímulos em variadas modalidades. Nunca temos acesso à Realidade, como ela é, e sim a uma reconstrução computacional das migalhas que conseguimos catar.

# Uma Pausa: Origens Evolutivas

As origens evolutivas dessas capacidades computacionais da vida são um problema em aberto.

![](https://64.media.tumblr.com/1b65c7a270ad1aa12f328598d54adc52/578a8a50523eba99-d8/s500x750/0a59e64b2d72f7cf0c47245583337859e0684b35.gifv)

> Essa amigável massaroca semi-transparente em formato de lágrima é da espécie _Lacrymaria Olor_. É um [protista ciliado](https://en.wikipedia.org/wiki/Ciliate) que caça extendendo um membro a distâncias de cerca de 5-6 vezes o seu próprio tamanho. Esse "bichinho" não tem um cérebro, mas dá conta de todas as suas necessidades metabólicas caçando. Mais ainda, estamos falando de um ser vivo unicelular. Mesmo nessa configuração extremamente simples existe fino controle motor, alguma capacidade de planejamento, de representação do estado do ambiente com algum grau de precisão (e.g. localização de comida relativa à posição atual). 

Author: ```suggestion
Essa amigável massaroca semi-transparente em formato de lágrima é da espécie _Lacrymaria Olor_. É um [protista ciliado](https://en.wikipedia.org/wiki/Ciliate) que caça extendendo um membro a distâncias de cerca de 5-6 vezes o seu próprio tamanho. Esse ser vivo não tem um cérebro verdadeiro, muito menos um sistema nervoso, mas dá conta de todas as suas necessidades - incluindo caça. Mais ainda, estamos falando de um ser vivo unicelular. Mesmo nessa configuração extremamente simples existe fino controle motor, alguma capacidade de planejamento, de representação do estado do ambiente com algum grau de precisão (e.g. localização de comida relativa à posição atual). 
```

Nas menores escalas da vida existe computação acontecendo para viabilizar _tudo_. O comportamento de uma Lacrymaria não é aleatório, ele persegue objetivos sob restrições energéticas. Note que esta humilde célula tem poder computacional para encontrar o comportamento apropriado em todas as modalidades necessárias à sua sobrevivência: anatômicas, metabólicas, fisiológicas, comportamentais, etc. 

O desenvolvimento da multicelularidade não jogou estas capacidades individuais no lixo, há um ganho de escala. Nossos cérebros não realizam nada de fundamentalmente diferente do que a nossa amiga Lacrymaria faz para funcionar. A capacidade de transmitir e encodar informação com diferenciais de voltagem é usada por ciliados unicelulares e cérebros de mamíferos para solucionar os problemas computacionais inerentes a navegar em um ambiente.

> Lacrymarias param de projetar seus membros ao alcançar comida, várias espécies regeneram membros inteiros em caso de amputação, nossos músculos hipertrofriam, nós sabemos quando resolvemos um problema de cálculo. Coletivos de células são capazes de perseguir objetos maiores, muito maiores, do que cada participante, em espaços de problemas completamente distintos. Nossa capacidade de navegar problemas sociais, linguísticos e econômicos está apoiada nos ombros de capacidades muito mais antigas e básicas, de navegar problemas comportamentais, anatômicos, fisiológicos, etc...

Author: ```suggestion
Lacrymarias param de projetar seus membros ao alcançar comida, várias espécies regeneram membros inteiros em caso de amputação, nossos músculos hipertrofriam, nós sabemos quando resolvemos um problema de cálculo. Coletivos de células são capazes de perseguir objetos maiores, muito maiores, do que cada participante, em espaços de problemas completamente distintos. Nossa capacidade de navegar problemas sociais, linguísticos e econômicos está apoiada nos ombros de capacidades muito mais antigas e básicas, de navegar problemas comportamentais, anatômicos, fisiológicos, moleculares, etc...
```

# Níveis de Marr

> Note que no nível mais elementar o seu funcionamento depende de física quântica (como você acha que se formam moléculas?) e no nível mais alto, de processos computacionais. Uma Teoria de Consciência precisa, simultaneamente, explicar quais são as funções seu corpo realiza, como elas viabilizam as capacidades de alto-nível que usamos a todo momento e também como isso é representado, computado e implementado no substrato biológico.
> 
> @marr1979computational oferece uma maneira de quebrar nossas perguntas (e respostas) em caixinhas com algum isolamento de responsabilidades. Nesse framework, qualquer sistema de processamento de informação pode ser, a princípio, entendido em 3 níveis:

Author: ```suggestion
Note que estamos rapidamente transitando entre escalas diferente, quase que por obrigação. Em um momento falamos de algo visível a olho nu, depois em células únicas, por vezes de sistemas inteiros, por vezes de propriedades matemáticas de certos algoritmos. Essa é uma fonte de problemas, é muito fácil começar uma frase em um contexto e terminar em outro, se confundindo no caminho. Uma Teoria de Consciência precisa, simultaneamente, explicar fenômenos em escalas espaciais, temporais e computacionais totalmente distintas e precisamos de algum _framework_ para evitar confusões básicas. @marr1979computational oferece uma maneira de quebrar nossas perguntas (e respostas) em escopos distintos que conversam naturalmente com nossos métodos. Nesse framework, qualquer sistema com aparente processamento de informação pode ser entendido em 3 níveis:
```
